source/_posts/2015-10-17-headless-firefox-in-ubuntu-on-virtual-box-for-cucumber-testing.markdown

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: "Headless Firefox in Ubuntu on VirtualBox for Cucumber testing"
+date: 2015-10-17 14:12:56 +0200
+comments: true
+author: Ivica Lakatoš
+categories: 
+  - Ruby and Rails
+  - Testing
+tags: 
+  - ruby
+  - rails
+  - vagrant
+  - virtualbox
+  - ubuntu
+  - cucumber
+  - selenium-webdriver
+published: true
+---
+
+If you use [Vagrant](http://www.vagrantup.com/downloads.html), [VirtualBox](https://www.virtualbox.org/) and Ubuntu to build your Rails apps and you want to test it with Cucumber scenarios, this is right post for you. By default Vagrant and VirtualBox use Ubuntu without an X server and GUI.
+
+Everything goes well until you need `@javascript` flag for your cucumber scenario. `@javascript` uses a javascript-aware system to process web requests (e.g. Selenium) instead of the default (non-javascript-aware) webrat browser.
+
+### Install Mozilla Firefox
+
+Selenium WebDriver is flexible and let you run selenium headless in servers with no display. But in order to run, Selenium needs to launch a browser. If there are no display to the machine, the browsers are not launched. So in order to use selenium, you need to fake a display and let selenium and the browser thinks they are running in a machine with a display.
```

Comment:
@Ikera just a few grammar issues: 
Selenium WebDriver is flexible and let -> Selenium WebDriver is flexible and **lets**
If there are no display -> If there **is** no **display**  OR If there **are** no **displays** 
let selenium and the browser thinks -> let selenium and the browser **think**